src/fs/file.rs

```diff
@@ -170,6 +171,56 @@ impl File {
         op.read().await
     }
 
+    /// Like [`read_at`], but using a pre-mapped buffer
+    /// registered with [`BufRegistry`].
+    ///
+    /// [`read_at`]: Self::read_at
+    /// [`BufRegistry`]: crate::buf::fixed::BufRegistry
+    ///
+    /// # Errors
+    ///
+    /// In addition to errors that can be reported by `read_at`,
+    /// this operation fails if the buffer is not registered in the
+    /// current `tokio-uring` runtime.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use tokio_uring::fs::File;
+    /// use tokio_uring::buf::fixed::BufRegistry;
+    /// use tokio_uring::buf::IoBuf;
+    /// use std::iter;
+    ///
+    /// fn main() -> Result<(), Box<dyn std::error::Error>> {
+    ///     tokio_uring::start(async {
+    ///         let registry = BufRegistry::new(iter::repeat(vec![0; 10]).take(10));
+    ///         registry.register()?;
+    ///
+    ///         let f = File::open("foo.txt").await?;
+    ///         let buffer = registry.check_out(2).unwrap();
+    ///
+    ///         // Read up to 10 bytes
+    ///         let (res, buffer) = f.read_fixed_at(buffer.slice(..), 0).await;
+    ///         let n = res?;
+    ///
+    ///         println!("The bytes: {:?}", &buffer[..n]);
+    ///
+    ///         // Close the file
+    ///         f.close().await?;
+    ///         Ok(())
+    ///     })
+    /// }
```

Comment:
Might be good to hide the details not specific to this example so user can just read and focus on the relevant details:

```suggestion
    ///# fn main() -> Result<(), Box<dyn std::error::Error>> {
    /// tokio_uring::start(async {
    ///     let registry = BufRegistry::new(iter::repeat(vec![0; 10]).take(10));
    ///     registry.register()?;
    ///
    ///     let f = File::open("foo.txt").await?;
    ///     let buffer = registry.check_out(2).unwrap();
    ///
    ///     // Read up to 10 bytes
    ///     let (res, buffer) = f.read_fixed_at(buffer.slice(..), 0).await;
    ///     let n = res?;
    ///
    ///     println!("The bytes: {:?}", &buffer[..n]);
    ///
    ///     // Close the file
    ///     f.close().await?;
    ///     Ok(())
    /// })
    ///# }
```

Reply:
Good idea! I copy-pasted this from the other examples, so perhaps they should be changed likewise.